scripts/linters/js_ts_linter_test.py

```diff
@@ -715,6 +719,68 @@ def mock_compile_all_ts_files():
             'the end of the comment.'], self.linter_stdout)
         self.assert_failed_messages_count(self.linter_stdout, 1)
 
+    def test_angular_services_index_error(self):
+        def mock_compile_all_ts_files():
+            cmd = (
+                './node_modules/typescript/bin/tsc -outDir %s -allowJS %s '
+                '-lib %s -noImplicitUseStrict %s -skipLibCheck '
+                '%s -target %s -typeRoots %s %s typings/*') % (
+                    js_ts_linter.COMPILED_TYPESCRIPT_TMP_PATH +
+                    'scripts/linters/test_files/', 'true', 'es2017,dom', 'true',
+                    'true', 'es5', './node_modules/@types',
+                    VALID_UNLISTED_SERVICE_PATH)
+            subprocess.call(cmd, shell=True, stdout=subprocess.PIPE)
```

Comment:
Why are we running commands in mock function?

Reply:
I am not really sure. This is how it was in other functions, Maybe @Hudda can answer this question better.

Reply:
@Hudda, is this expected? Do we really need this in the test? Can you please help @srijanreddy98?

Reply:
Yes we need it here because we are compiling a single file here instead of all files by mocking the compile method in js_ts_linter.py else tests become very slow.

Reply:
Can this be added as an explanatory comment?

Reply:
We are expecting the lint functions (check_angular_services_index) to be a public function. Considering that can you write a unit test for the specific method?

Reply:
> We are expecting the lint functions (check_angular_services_index) to be a public function. Considering that can you write a unit test for the specific method?

I am not sure what this means. Do I have to change something lines to look like: `js_ts_linter.check_angular_index_file()`?

Reply:
I think I will do this in a different pr. I have left a note at the top of the file explaining the subprocess call. I want this pr to go in before more contributors start making service migration PRs. Given that I have barely spent any time in the linter files, I won't be able to get an LGTM in one try and this would take two-three review cycles before it reaches that stage. So, I would like to do this in a different pr. Would that be ok @seanlip, @DubeySandeep?

Reply:
I'll defer to @DubeySandeep on this, since you've addressed my part of this comment thread.

Reply:
>  So, I would like to do this in a different pr. 

I'm fine with that, also we have an open issue to fix the linter test for js_ts_linter files at #10482. Thanks for explaining the reason so clearly @srijanreddy98! :)